src/main/kotlin/com/campspot/lib/PunchLib.kt

```diff
@@ -4,20 +4,23 @@ import com.campspot.api.Punch
 import com.campspot.dao.PunchDAO
 import com.campspot.exceptions.EntityNotFoundException
 import com.campspot.exceptions.PunchCannotOverlapException
+import com.campspot.middleware.transactions.DAOManager
 import java.time.ZonedDateTime
 
 class PunchLib(
-  private val punchDAO: PunchDAO,
+  private val daoManager: DAOManager,
   private val mockableObject: MockableObject
 ) {
   fun create(punch: Punch): Punch {
+    val punchDAO = daoManager[PunchDAO::class]
     validatePunchDoesNotOverlap(punch)
 
     val id = punchDAO.create(punch)
     return punchDAO.findById(id)
   }
 
   fun update(punch: Punch): Punch {
+    val punchDAO = daoManager[PunchDAO::class]
```

Comment:
There are a few different syntax choices for this, not that it is too important, I thought the square branckets looked nice since we will do this a lot.

Other options would be a generic invocation `daoManager<PunchDAO>()` or just a regular method call `daoManager.dao(PunchDAO::class)`